### PR TITLE
grid_sampler_2d: removed lowering

### PR DESCRIPTION
**Pull request by fdrocha**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #1134

There is now a decomposition in pytorch that seems to have
better performance, see benchmarks at
https://github.com/pytorch/pytorch/pull/84350

### CODE REVIEW

**Review by Chillee — APPROVED** [review]

File: torchinductor/lowering.py
Comment (Chillee) [inline torchinductor/lowering.py:930]: I don't think this PR resolves that, no? Since we're currently applying decompositions after autograd.

Comment (fdrocha) [reply torchinductor/lowering.py]: You are right! Thanks, fixed now.

**Comment by fdrocha** [comment]
The test failures are in `test/test_aot_cudagraphs.py`. They seem unrelated to this PR, I get them also on main.
@Chillee any thoughts?

**Comment by Chillee** [comment]
@fdrocha yeah they're not caused by this PR, but by updating the pin.

**Comment by fdrocha** [comment]
> @fdrocha yeah they're not caused by this PR, but by updating the pin.

Thanks @Chillee . I was able to get the tests to go green by replacing "fake_result" with "val" in `torchdynamo/torchdynamo/optimizations/training.py` and in `torch/fx/passes/backends/cudagraphs.py`.
I think you did something similar in https://github.com/pytorch/pytorch/pull/84331

Should I make those changes? Don't understand those parts of the code base, but it seems like the right thing...

There also a bunch of other instances of "fake_result" in torch repo that should probably be changed to "val" FWIW

**Comment by Chillee** [comment]
@fdrocha Yeah we just need to get a PR in updating all the uses (also see https://github.com/pytorch/pytorch/pull/84432).

**Comment by fdrocha** [comment]
> @fdrocha Yeah we just need to get a PR in updating all the uses (also see [pytorch/pytorch#84432](https://github.com/pytorch/pytorch/pull/84432)).

I see. I guess I will wait for that PR to be merged before trying to merge this one.

**Comment by jansel** [comment]
We could also just skip the tests until the upstream breaking is fixed.

**Comment by Chillee** [comment]
Yeah let's do that and land this.

**Comment by jansel** [comment]
This looks like it was not properly merged @fdrocha -- the merge button doesn't work with ghstack

**Comment by fdrocha** [comment]
> This looks like it was not properly merged @fdrocha -- the merge button doesn't work with ghstack

Drat. I did use `ghstack land` to merge this but I got a bunch of permission related errors. I assumed it was okay because GH says "successfully merged" but guess not. 

Any idea on how to fix this?

**Comment by Lezcano** [comment]
@ezyang what permissions do we need to land ghstack PRs?

**Comment by ezyang** [comment]
This is landed now. Apparently you need to be repo admin lol

**Comment by ezyang** [comment]
oops actualy not yet

**Comment by Lezcano** [comment]
@fdrocha we should start sending PRs without ghstack then :(

**Comment by fdrocha** [comment]
@ezyang this is not merged yet, right?
Do you think I should just make a new PR and merge that?

**Comment by fdrocha** [comment]
Ok, never mind then!

**Comment by Chillee** [comment]
I was talking to @ezyang and his opinion was that "we should just ask him to merge ghstack PRs if we want them landed" (for now).

**Pull request merged**
